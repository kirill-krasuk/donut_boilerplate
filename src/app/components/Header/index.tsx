import React    from 'react';
import loadable from '@loadable/component';

export default loadable(
    () => import(/* webpackChunkName: "Header" */'./component'),
    {
        fallback: <div>Loading...</div>
    }
);
