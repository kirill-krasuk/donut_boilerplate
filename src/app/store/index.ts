/* eslint-disable unicorn/prefer-module */
import { createStore } from '@lib/redux';
import { env }         from '@env/index';
import { Environment } from '@enums/env';
import rootReducer     from './reducers';
import appMiddlewares  from './middlewares';

const { appEnv } = env.client;

const preloadedState = __IS_CLIENT__
    ? (window as any).__PRELOADED_STATE__
    : {};

const store = createStore(
    rootReducer,
    preloadedState,
    appMiddlewares
)((store) => {
    if (__IS_CLIENT__ && appEnv === Environment.Dev && (module as any).hot) {
        (module as any).hot.accept('./reducers', () => store.replaceReducer(rootReducer as any));
    }
});

export { store };
