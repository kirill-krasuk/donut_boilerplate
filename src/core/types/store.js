// @flow

export type StoreType = {
    dispatch: Function,
    getState: Function
}
