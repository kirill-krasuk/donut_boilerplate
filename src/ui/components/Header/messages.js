import { defineMessages } from 'react-intl';

export default defineMessages({
    title: {
        id            : 'ui.components.Header.title',
        defaultMessage: 'Welcome to Donut Boilerplate v2'
    }
});
