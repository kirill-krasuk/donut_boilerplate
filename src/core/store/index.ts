import { configureStore } from '@core/utils/store/configureStore';

const preloadedState = __IS_CLIENT__
    ? (window as any).__PRELOADED_STATE__
    : {};

export const { store, rootReducer } = configureStore(preloadedState);
export { shakeReducers } from '@core/utils/store/shakeReducers';
