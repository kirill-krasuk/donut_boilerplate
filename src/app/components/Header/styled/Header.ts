import styled, { css } from 'styled-components/macro';

import { Theme }       from '@core/types/theme';

export const Header = styled.header`
    ${ ({ theme }: Theme) => css`
        width: 100%;
        background: ${ theme[theme.mode].header_bg };
        position: relative;
        display: flex;
        justify-content: center;

        &:before {
            background: ${ theme[theme.mode].secondary };
            content: '';
            height: 2px;
            width: 60%;
            position: absolute;
            bottom: 0;
            z-index: 10;
        }
    ` };
`;
