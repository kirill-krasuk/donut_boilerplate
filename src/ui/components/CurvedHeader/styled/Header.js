import styled            from 'styled-components/macro';

import { HEADER_HEIGHT } from 'ui/constants/header';

export default styled.header`
    height: ${ HEADER_HEIGHT };
    width: 100%;

    &:before {
        content: '';
        background: ${ ({ theme }) => `linear-gradient(90deg, ${ theme[theme.mode]['ui_ch_header--lg-from'] } , ${ theme[theme.mode]['ui_ch_header--lg-to'] })` };
        height: inherit;
        width: 100%;
        border-radius: 0 0 50% 50%;
        transform: scaleX(1.2);
        position: absolute;
        top: 0;
        left: 0;
        z-index: 1;
    }
`;
