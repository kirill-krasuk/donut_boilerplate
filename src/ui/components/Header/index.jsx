// @flow
import React                        from 'react';
import type { Node }                from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Sun }                      from 'styled-icons/fa-solid/Sun';
import { Moon }                     from 'styled-icons/fa-solid/Moon';

import { changeThemeAction }        from 'core/actions/theme';
import { getMode }                  from 'core/selectors/theme';
import * as Styled                  from './styled';

/**
 * @visibleName Header
 * @version 1.0
 * @author [Kirill Krasuk](https://github.com/kirill-krasuk)
 */

const Header = (): Node => {
    const dispatch = useDispatch();

    /**
     * theme mode variable
     * @enum {string}
    */

    const mode = useSelector(getMode);

    const ThemeIcon = {
        light: Sun,
        dark : Moon
    }[mode];

    const ThemeLogo = {
        dark: Styled.Logo,
        light : Styled.DarkLogo
    }[mode];

    const handleChangeTheme = () => {
        const reverseMode = mode === 'light'
            ? 'dark'
            : 'light';

        dispatch(changeThemeAction(reverseMode));
    };

    return (
        <Styled.Header>
            <Styled.Nav>
                <Styled.ThemeIcon>
                    <ThemeIcon onClick={ handleChangeTheme } />
                </Styled.ThemeIcon>
            </Styled.Nav>
            <Styled.Content>
                <ThemeLogo />
                <Styled.Title>Welcome to Donut Boilerplate v2</Styled.Title>
            </Styled.Content>
        </Styled.Header>
    );
};

export default Header;
