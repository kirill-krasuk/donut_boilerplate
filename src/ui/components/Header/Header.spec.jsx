import React        from 'react';
import { Provider } from 'react-redux';

import { store }    from 'test/__mocks__/store';
import Header       from '.';

describe('UI Header component', () => {
    describe('Header init', () => {
        const header = shallow(
            <Provider store={ store }>
                <Header />
            </Provider>
        );

        it('renders properly', () => {
            expect(header).toMatchSnapshot();
        });
    });
});
