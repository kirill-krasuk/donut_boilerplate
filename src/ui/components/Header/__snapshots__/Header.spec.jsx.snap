// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI Header component Header init renders properly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    }
  }
>
  <Header />
</ContextProvider>
`;
